uradvd: update to r26-1e64364d

AiyionPrime opened the pull request:
## 📦 Package Details

**Maintainer:** @mwarning 

**Description:**
Implement a new option:
--version

Provide a repository overview in the readme.

---

## 🧪 Run Testing Details

- **OpenWrt Version:** main
- **OpenWrt Target/Subtarget:** x86
- **OpenWrt Device:** 64

---

## ✅ Formalities

- [x] I have reviewed the [CONTRIBUTING.md](https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md) file for detailed contributing guidelines.

### If your PR contains a patch:

- ~It can be applied using `git am`~
- ~It has been refreshed to avoid offsets, fuzzes, etc., using~
  ```bash
  make package/<your-package>/refresh V=s
  ```
- ~It is structured in a way that it is potentially upstreamable~
<sub>(e.g., subject line, commit description, etc.)</sub>
<sub>We must try to upstream patches to reduce maintenance burden.</sub>

wehagy commented:
`PKG_RELEASE` and `PKG_VERSION` are two different things and shouldn't be tied together, or you may be asking for trouble.

Why reinvent the wheel? Is the default `PKG_SOURCE_DATE` not good enough?

wehagy commented on net/uradvd/Makefile:
```diff
@@ -5,9 +5,9 @@ PKG_RELEASE:=1
-PKG_MIRROR_HASH:=1f681e2baf6f9b825668aeed9c075f532db509d7d1848e76f3e8101acf2bd476
+PKG_SOURCE_VERSION:=1e64364d323acb8c71285a6fb85d384334e7007d
+PKG_MIRROR_HASH:=f16519b85332c03c3147f7eaecdc656186d5ae530b0ad550b48a42f39d40acc7
+PKG_VERSION:=26~1e64364d
```
```suggestion
PKG_VERSION:=26~$(shell t=$(PKG_SOURCE_VERSION); echo $${t:0:8})
```

AiyionPrime commented:
Ah, I think I got aparcar wrong.
I aimed for consuming the hash and use it in the version like @wehagy suggested.
Thanks you two.

@mwarning Is there anything left?

wehagy commented:
I don't like the idea of using a custom version, but the maintainer decides, and he/she approved it, so let's go.

Also, according to the [Submission Guidelines](https://openwrt.org/submitting-patches#submission_guidelines), don't capitalize the first word after the prefix in the commit subject.

After fixing the commit subject, I will merge it. Thanks.

AiyionPrime commented:
rebased onto master

AiyionPrime commented:
My bad, capitalizing after the colon is a habit I took on using conventional commits.

I fixed the title and found one arch to be failing.
I rebased onto master, but the problem remains.
It appears to be unrelated to the PR though.

BKPepe commented:
> Why reinvent the wheel? Is the default `PKG_SOURCE_DATE` not good enough?

I am with @wehagy on this one. Why do we need to have custom version?
What about using semantic versioning in upstream together with tagged releases? 

From the commit itself, I can not see why it was switched from PKG_SOURCE_DATE to PKG_VERSION. :-(

AiyionPrime commented:
Thanks for your patience, everyone.

As I'd like to use `uradvd` on a nix machine, I packaged it a few days ago.
Lacking an official version scheme, the package repos had different fallback schemes.

OpenWrt demanded
`2021-09-14-9b0da60e` in 23.05 and
`2025.08.16~b37524df` in 24.10 for unversioned projects,
while nix demands something like this:
`0-unstable-2025-08-16`.

Looking for a scheme which provided a better overview for systematic reviews,
I suggested an official version.

I spare the detail, but the gist of the evening-fillign discussion was:
- versions are possible
- the uptream author dislikes both
  - semver for the project as well as
  - anything that included dates in general
- and instead favored the scheme, which you now see,
- as well as any form of tags, which would duplicate version informations of his scheme.

I must say that I'm quite happy with having an official version at all,
and that I'm sorry, if you are unhappy with the chosen system.

While I did not invent this wheel, I definitely led to its development.

Anyway, the project `uradvd` has an official version now,
the projects author made clear that he does not want `uradvd` to take packaging repos version desires into account,
I tried, when I encountered APK.

I hope OpenWrt does not insist on the fallback version scheme for `uradvd`.

Good night everyone.

PS: I added a sentence about `uradvd` now having an official version scheme in the commit message to take @BKPepe's suggestion into account.

AiyionPrime commented:
Thanks for the answer;
I'll follow up properly, when we're down again tonight, but what exactly is the normal update process?

The hash of the revision must come from somewhere, is that a manual inspection of the repo, or is there a script?

1715173329 commented on net/uradvd/Makefile:
```diff
@@ -5,9 +5,9 @@ PKG_RELEASE:=1
-PKG_MIRROR_HASH:=1f681e2baf6f9b825668aeed9c075f532db509d7d1848e76f3e8101acf2bd476
+PKG_SOURCE_VERSION:=1e64364d323acb8c71285a6fb85d384334e7007d
+PKG_MIRROR_HASH:=f16519b85332c03c3147f7eaecdc656186d5ae530b0ad550b48a42f39d40acc7
+PKG_VERSION:=26~$(shell t=$(PKG_SOURCE_VERSION); echo $${t:0:8})
```
```suggestion
PKG_VERSION:=26~$(call version_abbrev,$(PKG_SOURCE_VERSION))
```

1715173329 replied:
Yup.
See https://github.com/openwrt/openwrt/blob/master/rules.mk#L86

1715173329 commented on net/uradvd/Makefile:
```diff
@@ -25,6 +25,10 @@ define Package/uradvd/description
 	 Advertise an IPv6 prefix/route via SLAAC.
 endef
 
+define Build/Compile
+	$(call Build/Compile/Default,VERSION=r$(subst ~,-,$(PKG_VERSION)))
+endef
```
```suggestion
MAKE_FLAGS += VERSION=r$(subst ~,-,$(PKG_VERSION))
```

try to simplify it with MAKE_FLAGS

1715173329 commented:
Honestly I still don't figure out why we need a `r26` prefix. I cannot even find this magic number in upstream repo.

1715173329 commented:
Thanks for the explanation, and I found wehagy already answered that, sorry for the extra noise.

It might be good to add some comments above `PKG_VERSION` to describe the versioning to make it a bit clear.

BKPepe commented:
Hey @neocturne could you please check recent comments and tell us, why you need to have some special versioning? 🤔 
It looks like you are trying to reinvent a wheel here and we as said by in very well explained answer https://github.com/openwrt/packages/pull/27482#issuecomment-3323157124, it is just weird that upstreaming versioning is not exactly user-friendly for us.

mwarning commented:
@AiyionPrime I would say that we should not try to access to git data. In my opinion, the git data is not part of the source code and try to access it is a additional dependency that will lead to trouble. 
As such, I would proceed with setting the version manually.
